homeassistant/components/blink/config_flow.py

```diff
@@ -0,0 +1,109 @@
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        stored_username = ""
+        stored_password = ""
+        stored_interval = DEFAULT_SCAN_INTERVAL
+
+        if BLINK_CONFIG in self.hass.data:
+            stored_username = self.hass.data[BLINK_CONFIG].get(CONF_USERNAME)
+            stored_password = self.hass.data[BLINK_CONFIG].get(CONF_PASSWORD)
+            stored_interval = self.hass.data[BLINK_CONFIG].get(CONF_SCAN_INTERVAL)
+
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[CONF_USERNAME])
+
+            if CONF_SCAN_INTERVAL not in user_input:
+                user_input[CONF_SCAN_INTERVAL] = stored_interval
+
+            try:
+                self.blink = self.hass.data[DOMAIN]
+            except KeyError:
+                self.blink = blinkpy.Blink(
+                    username=user_input["username"],
+                    password=user_input["password"],
+                    motion_interval=DEFAULT_OFFSET,
+                    legacy_subdomain=False,
+                    no_prompt=True,
+                    device_id="Home Assistant",
+                )
+                self.blink.refresh_rate = user_input[CONF_SCAN_INTERVAL]
+                await self.hass.async_add_executor_job(self.blink.start)
+
+            if not self.blink.key_required:
+                # No key required, we're good
+                self.hass.data[DOMAIN] = self.blink
+                config_data = {
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                    CONF_SCAN_INTERVAL: user_input[CONF_SCAN_INTERVAL],
+                }
+                return self.async_create_entry(title=DOMAIN, data=config_data,)
+            return await self.async_step_2fa()
+
+        data_schema = {
+            vol.Required("username", default=stored_username): str,
+            vol.Required("password", default=stored_password): str,
+        }
+
+        if self.show_advanced_options:
+            data_schema[vol.Required("scan_interval", default=stored_interval)] = int
+
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema(data_schema), errors={},
+        )
+
+    async def async_step_2fa(self, user_input=None):
+        """Handle 2FA step."""
+        if user_input:
+            pin = user_input[CONF_PIN]
+            if not pin or pin.lower() == "none":
+                pin = None
+            if self.blink.login_handler.send_auth_key(self.blink, pin):
+                await self.hass.async_add_executor_job(self.blink.setup_post_verify)
+                self.hass.data[DOMAIN] = self.blink
+                config_data = {
+                    CONF_USERNAME: self.blink.login_handler.data["username"],
+                    CONF_PASSWORD: self.blink.login_handler.data["password"],
```

Comment:
This seems very weird. We need to store the username and password but won't get 2FA challenges anymore ? 

I would expect we can exchange username + password + 2fa for a token and then no longer have to touch username and password at all.

Reply:
Does this mean people need to open a link in their email after they restart Home Assistant ?